### PR TITLE
Rework spec tags

### PR DESCRIPTION
**Pull request by nforro**

### CODE REVIEW

File: rebasehelper/spec_content.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
```
Comment (FrNecas) [inline rebasehelper/spec_content.py]: You should also probably move its tests to a separate file

File: rebasehelper/specfile.py [hunk 2]
```diff
@@ -605,7 +529,7 @@ def set_extra_version(self, extra_version: Optional[str], version_changed: bool)
         # TODO: in some cases it might be necessary to modify Source0
 
     @saves
-    def set_tag(self, tag: str, value: str, preserve_macros: bool = False, subpackage: Optional[str] = None) -> None:
```
Comment (FrNecas) [inline rebasehelper/specfile.py]: Why don't we need the subpackage? What if we wanted to alter a tag in a subpackage, this wouldn't be possible after the removal, right?

Comment (nforro) [reply rebasehelper/specfile.py]: Feel free to adjust `set_tag()` to work with subpackages, but I don't think it's worth it. What would be the use-case?

Comment (FrNecas) [reply rebasehelper/specfile.py]: Makes sense, there is probably no point in having `set_tag` work with subpackages, at least for now

File: rebasehelper/spec_content.py (hunk 1 above)
Comment (nforro) [reply rebasehelper/spec_content.py]: Done. I've also added tests for Tags.

File: rebasehelper/tests/test_spec_content.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+#          Nikola Forró <nforro@redhat.com>
+#          František Nečas <fifinecas@seznam.cz>
+
+import pytest  # type: ignore
+
+from rebasehelper.spec_content import SpecContent
+from rebasehelper.tests.conftest import SPEC_FILE
+
+
+class TestSpecContent:
```
Comment (FrNecas) [inline rebasehelper/tests/test_spec_content.py]: There are also public API tests for SpecContent. Although they are quite short, I think they should be moved as well for the sake of consistency.

**Pull request merged**